src/DynamoCore/Utilities/LuceneSearchUtility.cs

```diff
@@ -368,153 +373,151 @@ internal string CreateSearchQuery(string[] fields, string SearchTerm, bool IsPac
-                        {
-                            wildcardQuery.Boost = LuceneConfig.WildcardsSearchNameParsedWeight;
-                        }
-                        else
-                        {
-                            wildcardQuery.Boost = LuceneConfig.FuzzySearchWeight;
-                        }
-                        booleanQuery.Add(wildcardQuery, Occur.SHOULD);
+                        //Adds the FuzzyQuery and 4 WildcardQueries (3 of them contain regular expressions), with the weights for Queries with RegularExpressions
+                        AddQueries(s, f, searchType, booleanQuery, occurQuery, LuceneConfig.FuzzySearchMinEdits, true);
                     }
                 }
             }
             return booleanQuery.ToString();
         }
 
-        private WildcardQuery CalculateFieldWeight(string fieldName, string searchTerm, bool isWildcard = false)
+        /// <summary>
+        ///  //Adds the FuzzyQuery and 4 WildcardQueries (3 of them contain regular expressions) with specific weight for each one
+        /// </summary>
+        /// <param name="searchTerm">Search Term introduced by the user</param>
+        /// <param name="field">Field being processed</param>
+        /// <param name="searchType">Type of Search: Normal, ByDotCategory and ByEmptySpace</param>
+        /// <param name="booleanQuery">The Boolean query in which the Wildcard queries will be added</param>
+        /// <param name="occurQuery">Occur type can be Should or Must</param>
+        /// <param name="fuzzyLogicMaxEdits">Max edit lenght for Fuzzy queries</param>
+        /// <param name="termSplit">Indicates if the SearchTerm has been split by empty space or not</param>
+        private void AddQueries(string searchTerm, string field, SearchType searchType, BooleanQuery booleanQuery, Occur occurQuery, int fuzzyLogicMaxEdits, bool termSplit = false)
+        {
+            string querySearchTerm = searchTerm.Replace(" ", string.Empty);
+
+            FuzzyQuery fuzzyQuery;
+            if (searchTerm.Length > LuceneConfig.FuzzySearchMinimalTermLength)
+            {
+                fuzzyQuery = new FuzzyQuery(new Term(field, querySearchTerm), fuzzyLogicMaxEdits);
+                booleanQuery.Add(fuzzyQuery, Occur.SHOULD);
+            }
+
+            if (searchType == SearchType.ByDotCategory && field == nameof(LuceneConfig.NodeFieldsEnum.CategorySplitted))
+            {
+                occurQuery = Occur.MUST;
+            }
+
+            foreach (WildcardType enumVal in Enum.GetValues(typeof(WildcardType)))
+            {
+                var wildcardQuery = CalculateFieldWeight(field, querySearchTerm, enumVal, termSplit);
+                booleanQuery.Add(wildcardQuery, occurQuery);
+            }
+        }
+
+        private WildcardQuery CalculateFieldWeight(string fieldName, string searchTerm, WildcardType wilcardType = WildcardType.None, bool termSplit = false)
```

Comment:
Can you add some comments here about the weight offset?

Reply:
Added more comments for this method, please let me know if after adding the comments now is more clear, see commit: https://github.com/DynamoDS/Dynamo/pull/15473/commits/40697ac1b0752b02a53c5ccb86b77a1a67747492